netdeploy/remote/nodecfg/nodeConfigurator.go

```diff
@@ -150,20 +157,30 @@ func (nc *nodeConfigurator) prepareNodeDirs(configs []remote.NodeConfig, rootCon
 		}
 
 		// Copy the bootstrapped files into current ledger folder
-		if nc.bootstrappedBlockFile != "" && nc.bootstrappedTrackerFile != "" {
+		if nc.bootstrappedBlockFile != "" &&
+			(nc.bootstrappedTrackerFile != "" || nc.bootstrappedTrackerDir != "") {
 			fmt.Fprintf(os.Stdout, "... copying block database file to ledger folder ...\n")
 			dest := filepath.Join(nodeDest, genesisDir, fmt.Sprintf("%s.block.sqlite", config.LedgerFilenamePrefix))
 			_, err = util.CopyFile(nc.bootstrappedBlockFile, dest)
 			if err != nil {
 				return nil, fmt.Errorf("failed to copy database file %s from %s to %s : %w", "bootstrapped.block.sqlite", filepath.Dir(nc.bootstrappedBlockFile), dest, err)
 			}
-			fmt.Fprintf(os.Stdout, "... copying tracker database file to ledger folder ...\n")
-			dest = filepath.Join(nodeDest, genesisDir, fmt.Sprintf("%s.tracker.sqlite", config.LedgerFilenamePrefix))
-			_, err = util.CopyFile(nc.bootstrappedTrackerFile, dest)
-			if err != nil {
-				return nil, fmt.Errorf("failed to copy database file %s from %s to %s : %w", "bootstrapped.tracker.sqlite", filepath.Dir(nc.bootstrappedBlockFile), dest, err)
+			if nc.bootstrappedTrackerFile != "" {
+				fmt.Fprintf(os.Stdout, "... copying tracker database file to ledger folder ...\n")
+				dest = filepath.Join(nodeDest, genesisDir, fmt.Sprintf("%s.tracker.sqlite", config.LedgerFilenamePrefix))
+				_, err = util.CopyFile(nc.bootstrappedTrackerFile, dest)
+				if err != nil {
+					return nil, fmt.Errorf("failed to copy database file %s from %s to %s : %w", "bootstrapped.tracker.sqlite", filepath.Dir(nc.bootstrappedBlockFile), dest, err)
+				}
+			}
+			if nc.bootstrappedTrackerDir != "" {
```

Comment:
should not `nc.bootstrappedTrackerFile != ""` and `nc.bootstrappedTrackerDir != ""` be mutually exclusive?

Reply:
they should be, but:

* For that to happen, a generated config would first need to contain both DB types.
* If both *are* present, we could take only one, or both, or none. As written now, it takes both, which should be safe since a running algod is only going to be using one of the two resources per its StorageEngine value. We could prefer one over the other, but we'd have to introspect somehow to know which is appropriate. Or we could take none, and error. Erroring is fine, but I don't think both DBs being present in a config is immediately cause for alarm.